### PR TITLE
Add fallback for disabled WebGL

### PR DESCRIPTION
**Pull request by MassiveBox**
Basically #5 but with actual fallbacks: if you have WebGL disabled, you'll get two pre-rendered GIFs instead of the actual 3D models.
The GIFs are not the best, but they do the job for devices and browsers without WebGL. They look decent, and most importantly loop right.

Again please let me know if I made some mistake. I've tested this on desktop and mobile browsers with and without WebGL, and everything works as expected, as far as my tests go.

I'll tag @Bilb real quick here since the last pull request slid under their radar 😉

### CODE REVIEW

**Comment by yougotwill** [comment]
Hi @MassiveBox thanks for contributing! I work on the oxen websites along with Bilb. I've going through your PR when I have time and will leave some comments. Will message here again when I have finished going through it.

**Comment by MassiveBox** [comment]
Hey @yougotwill, I'm pinging you again to discuss about acceptable file sizes for the fallbacks.  
I learnt how [unoptimzied](https://kornel.ski/efficient-gifs) GIFs are, and I agree that using videos is overkill, so I'd happily go with a static image. At the end of the day it's an almost meaningless decoration, so it doesn't matter as long as the website works.
So, one of the main requisites here is keeping the file sizes low, while retaining a decent quality. I understand that is important to keep the website accessible to all users.

I'm asking here what do you think the maximum image size should be for the wireframes. I've obtained some high quality screenshots of the animations, and I've edited them to remove the background, scale them and I've added some lossless compression. I've also migrated from using the Image tag with an explicit source to using imports (as recommended on the [docs](https://nextjs.org/docs/basic-features/image-optimization#using-the-image-component) for local images), so the images should be further optimized by Next.JS. So far I'm only testing on the wireframe sphere, and I now have a 500x506 still picture with a good quality. 
But as of now, by using the dev tools tab of my browser, I can see the sphere wireframe weights about 95KB. Much better than 7.3MB, but I'm afraid it still might be seen as too high, that's why I'm asking if it's good enough or not.

I'll attach the partial screenshots zipped below:  
[wireframes things.zip](https://github.com/oxen-io/lokinet-website/files/7446968/wireframes.things.zip)
Thank you :D

**Comment by yougotwill** [comment]
Hi @MassiveBox, Glad we are on the same page.

I think your approach has been very thorough. For the record what I normally do is run the image through ImageOptim (mac only but there are similar programs for windows or linux https://imageoptim.com/mac) and then use https://tinypng.com .

If you want to get a really small file size JPEGS are usually the way to go if you don't care about transparency. Since this site is always a white or black background we don't need to worry about it. Try changing the file format and see if that improves the file sizes you get.

Next regarding using imports, I usually still use the explicit source because we can then take advantage of `next/image` lazy loading the image. This helps increase the speed of the website the first time it is loaded on  someone's machine by loading images slightly later. You can check the first time loading sizes when you run `yarn build`. If we import the picture directly it means the site will be a bit slower on the first time it loads.

Keep up the great work. I've chatted with my colleagues and if you are interested in contributing to our other sites I can set some things up for you. No pressure though 😄

**Comment by MassiveBox** [comment]
Okay, I'll do some testing tomorrow and commit.  
Thank you again for your support. At first I was skeptical about the import stuff because I thought the docs must have told the best way, but a couple lighthouse audits confirmed you're right! Good call.  
About the other stuff, it would love to contribute if needed! I'm always trying to be helpful, but do keep in mind I don't have much free time, so I can't guarantee anything. I'll try my best :)

**Comment by MassiveBox** [comment]
Here @yougotwill, it's done, let me know if this is ok!
If you're wondering about https://github.com/oxen-io/lokinet-website/pull/6/commits/42f8cae5a90be0d74ca91639aa1676593508b723#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L22, I changed this because I was getting an error when installing dependencies, and by looking at the [NPM page](https://www.npmjs.com/package/@types/styled-components) it appears that the 5.2.3 version does not exist.  
I've tested everything I could, but let me know if there's something that still needs to be fixed!  
I did a couple of lighthouse audits, and without WebGL I got 96 performance, 83 accessibility and 80 best practices; with WebGL I got 98/81/80, so I guess it's not that worse :)

**Comment by yougotwill** [comment]
Hi @MassiveBox Sorry for the late reply. Been a bit swamped lately. I'll check this out when I can and get back to you! 

Appreciate you checking out the lighthouse stuff saves me doing it 😄

**Comment by yougotwill** [comment]
@MassiveBox I've checked it out. All looks great! I've commented on 1 small thing. As soon as you fix that I will merge and deploy it.

**Comment by yougotwill** [comment]
I started making our other sites more contributor friendly. I need some more time though. When I'm done I'll tag you in a good first issue on one of the projects. No pressure to do them if you aren't interested though. 👍

**Comment by MassiveBox** [comment]
Sorry for the newbie question @yougotwill, but I can't find the comment you're referring to. I've checked all the commits, my notifications tab, my mail and the GitHub Conversations for each commit. Could you please tell me where to find it?

**Comment by yougotwill** [comment]
Hi @MassiveBox no problem at all. I tagged you in the comment if that didn’t go to files changed and the go to components/WireFrameFallbacks.tsx

File: components/WireframeFallbacks.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import Image from "next/image";
```
Comment (MassiveBox) [inline components/WireframeFallbacks.tsx:1]: this is a test comment that I will soon delete

Comment (MassiveBox) [inline components/WireframeFallbacks.tsx:1]: This is a test comment ⭕ ❌

File: components/WireframeFallbacks.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { ThemeContext } from "../theme/theme";
```
Comment (MassiveBox) [inline components/WireframeFallbacks.tsx:2]: Test comment

**Comment by MassiveBox** [comment]
image
Still no luck, I feel dumb :S
Comments are on, yet I see none. I haven't even received the usual notification for being tagged somewhere. I've tried making a test comment and I can see it, but can you see mine?

**Comment by yougotwill** [comment]
That's very strange! Maybe you are right but you saw my earlier comments when you first made this PR right? Anyways here is a video of what I see for what it's worth.

https://user-images.githubusercontent.com/14887287/141700851-1fcbc275-2741-4240-bdf9-d9ccdc308495.mp4

If you can't find the comment all I need you to do is change the `export var` to `export const` in `WireframeFallbacks.tsx`.

**Comment by MassiveBox** [comment]
Alright @yougotwill, that very difficult change is done X)  
Also I've renamed white to light because consistency is always good.  
Lastly I can confirm I wasn't looking in some weird place, I was following the same path to see the comment(s), but I still see only mine. This needs to be figured out for the future, because inline comments seem to be very useful.

So, if there's nothing to change as you said, this is it for now :D  
Many commits for a quite simple fix, but I'm very thankful you took the time to explain me all the best practices. I've learnt so much here, and I'm sure it will come in handy for my future projects!  
Ping me if my collaboration is needed again. Thanks a lot :))

**Comment by yougotwill** [comment]
LGTM 🚀

**Comment by yougotwill** [comment]
@MassiveBox Thanks for your patience man! Not many people would put up with my nitpicking but I'm glad you saw the benefits. I'll be sure to tag you in future issues across our GitHub projects 😉 

I'll also try and fix the inline comments issue once I have full access to the repo.

**Pull request merged**